pkg/controller/directvolumemigration/rsync.go

```diff
@@ -79,623 +66,436 @@ const (
+					newOperation.Failed = currentStatus.failed
+					newOperation.Succeeded = currentStatus.succeeded
+					if newOperation.IsComplete() {
+						t.Log.Info(
+							fmt.Sprintf("Rsync operation completed after %d attempts", newOperation.CurrentAttempt),
+							"pvc", newOperation, "failed", newOperation.Failed, "succeded", newOperation.Succeeded)
+					} else {
+						t.Log.Info("Rsync operation is still running. Waiting for completion",
+							"pod", path.Join(pod.Namespace, pod.Name),
+							"pvc", newOperation,
+						)
+					}
+				}
+			} else {
+				newOperation.CurrentAttempt = 0
+				labels[RsyncAttemptLabel] = fmt.Sprintf("%d", currentStatus.operation.CurrentAttempt+1)
+				rsyncOptions = append(rsyncOptions, rsync_transfer.WithSourcePodLabels(labels))
+				transfer, err := rsync_transfer.NewTransfer(
+					stunnelTransport, endpoint, srcClient.RestConfig(), destClient.RestConfig(), pvcList, rsyncOptions...)
+				if err != nil {
+					currentStatus.AddError(err)
+					continue
+				}
+				err = transfer.CreateClient(srcClient)
+				if err != nil {
+					currentStatus.AddError(err)
+					continue
+				}
+			}
+			statusList.Add(currentStatus)
+			t.Log.Info("adding status of pvc", "pvc", currentStatus.operation, "errors", currentStatus.errors)
+		}
+	}
+	return statusList, nil
 }
 
-// Create rsync transfer route
-func (t *Task) createRsyncTransferRoute() error {
+func (t *Task) areRsyncTransferPodsRunning() (arePodsRunning bool, nonRunningPods []*corev1.Pod, e error) {
 	// Get client for destination
 	destClient, err := t.getDestinationClient()
 	if err != nil {
-		return err
+		return false, nil, err
 	}
+
 	pvcMap := t.getPVCNamespaceMap()
 	dvmLabels := t.buildDVMLabels()
 	dvmLabels["purpose"] = DirectVolumeMigrationRsync
+	selector := labels.SelectorFromSet(dvmLabels)
 
 	for bothNs, _ := range pvcMap {
 		ns := getDestNs(bothNs)
-		svc := corev1.Service{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      DirectVolumeMigrationRsyncTransferSvc,
-				Namespace: ns,
-			},
-			Spec: corev1.ServiceSpec{
-				Ports: []corev1.ServicePort{
-					{
-						Name:       DirectVolumeMigrationStunnel,
-						Protocol:   corev1.ProtocolTCP,
-						Port:       int32(2222),
-						TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: 2222},
-					},
-				},
-				Selector: dvmLabels,
-				Type:     corev1.ServiceTypeClusterIP,
-			},
-		}
-		svc.Labels = t.Owner.GetCorrelationLabels()
-		svc.Labels["app"] = DirectVolumeMigrationRsyncTransfer
-
-		t.Log.Info("Creating Rsync Transfer Service for Stunnel connection "+
-			"on destination MigCluster ",
-			"service", path.Join(svc.Namespace, svc.Name))
-		err = destClient.Create(context.TODO(), &svc)
-		if k8serror.IsAlreadyExists(err) {
-			t.Log.Info("Rsync transfer svc already exists on destination",
-				"service", path.Join(svc.Namespace, svc.Name))
-		} else if err != nil {
-			return err
+		pods := corev1.PodList{}
```

Comment:
probably worth using this? https://github.com/konveyor/crane-lib/blob/e077c6bf9b666c3aaa85946e8e924ebf0c632dc7/state_transfer/transfer/transfer.go#L33

Reply:
actually I think there are log message that we need for MTC, if you also determine that IsServerHealthy is not usable probably worth filing an issue we can circle back later

Reply:
@alaypatel07 I thought about using `IsHealthy` function. However, we are losing some advanced error heuristics by doing that. I propose to keep this function as-is for 1.6. WDYT?

Reply:
yes, works with me.